newIDE/app/src/ResourcesList/ResourceSelector.js

```diff
@@ -169,15 +169,16 @@ export default class ResourceSelector extends React.Component<Props, State> {
     const resourcesManager = project.getResourcesManager();
     const initialResource = resourcesManager.getResource(resourceName);
 
-    let initialResourceMetadata = {};
+    let externalEditorData = {};
```

Comment:
I think `initialResourceMetadata` was a good name and we don't have to change it to externalEditorData.
Let me explain:

initialResourceMetadata means "the metadata that were originally attached to the resource". This is perfectly valid and there is nothing related to external editor.
The external editor (jfxr or piskel) should get this info, but this should still be named initialResourceMetadata, because it's still the metadata.

BUT when piskel or jfxr are calling back `onChangesSaved`, they now pass an object which is called `externalEditorData` because this object is the "data that is passed from the external editor" and which CAN be used by GD to do something... like storing these informations in metadata ;)

But it's important to keep "metadata" as a term only for GD.

Reply:
I see now what you mean. I will rename the variables

Reply:
If we get this correct, all these things will be clearer and useful for other potential external editors in the future.

Reply:
yes I agree with you - in making it as clear as possible for others to be able to contribute easier

Reply:
amended this as requested